lib/SearchDescription.php

```diff
@@ -522,26 +523,30 @@ private function queryNearbyPoi(&$oDB, $iLimit)
 
         $aDBResults = array();
         $sPoiTable = $this->poiTable();
+        $sIndexName=$this->poiIndex();
+        $bIndex = $oDB->indexExists($sIndexName);
```

Comment:
Checking if an index exists during every search is slow. Better to check once (`$oDB->getListOfIndices()`), e.g. in a __construct, and cache the result.

Reply:
Unfortunately, we can't do this kind of caching because PHP is stateless. So, you'd recreate the list of indexes on every request. 

Looking forward to switching to Python exactly for those kind of reasons.

Reply:
Is Switching to Python on a visible roadmap somewhere?